File: Assets/zfoocs/Http/HttpHelloRequest.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class HttpHelloRequest
    {
        public string message;
    }

    public class HttpHelloRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1700;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            HttpHelloRequest message = (HttpHelloRequest) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            HttpHelloRequest packet = new HttpHelloRequest();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/http/HttpHelloRequest.lua

local HttpHelloRequest = {}

function HttpHelloRequest:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function HttpHelloRequest:protocolId()
    return 1700
end

function HttpHelloRequest:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function HttpHelloRequest:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = HttpHelloRequest:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return HttpHelloRequest
